### PR TITLE
Dbp 549 configure and separate configs and secrets in spsh helm chart keycloak

### PR DESCRIPTION
**Pull request by M-Schiborr**
# Description

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml [hunk 1]
```diff
@@ -56,19 +56,34 @@ spec:
             - name: KEYCLOAK_ADMIN
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.secrets.name }}
+                  {{- if .Values.auth.existingSecret }}
+                  name: {{ .Values.auth.existingSecret }}
+                  {{- end }}
+                  {{- if not .Values.auth.existingSecret }}
+                  name: {{ .Values.auth.name }}
```
Comment (simoncolincap) [inline charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:63]: In the secret template, the name is set to .Release.Name, not .Values.auth.name

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 2]
```diff
@@ -16,6 +16,18 @@ database:
 
 namespace: ""
 
+auth:
+  # existingSecret: Refers to a secret already present in the cluster, which is required for the authentication and configuration of the database setup tasks.
+  existingSecret: ""
+  name: keycloak
+  admin_password: ""
+  admin_user: ""
+  db_host: ""
```
Comment (simoncolincap) [inline charts/dbildungscloud-iam-keycloak/values.yaml:25]: I think the database credentials would make more sense under database

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml (hunk 1 above)
Comment (M-Schiborr) [reply charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:63]: Fixed that.

File: charts/dbildungscloud-iam-keycloak/values.yaml (hunk 2 above)
Comment (M-Schiborr) [reply charts/dbildungscloud-iam-keycloak/values.yaml]: moved them under database

**Review by simoncolincap — APPROVED** [review]

**Review by dimapin — APPROVED** [review]

**Pull request merged**